codec/codec.go

```diff
@@ -157,21 +169,23 @@ func NewClientCodec(conn net.Conn) *ClientCodec {
 // WriteRequest writes the appropriate header protobuf and the given protobuf
 // to the connection (each prefixed with a uvarint indicating its size).
 func (c *ClientCodec) WriteRequest(req *rpc.Request, obj interface{}) error {
-	pb, ok := obj.(proto.Message)
-	if !ok {
-		return fmt.Errorf("%T does not implement proto.Message", obj)
-	}
+	pb, hasMessage := obj.(proto.Message)
 
 	// Write the header
 	header := wire.Header{
 		Method: &req.ServiceMethod,
 		Seq:    &req.Seq,
 	}
+
 	if err := WriteProto(c.w, &header); err != nil {
 		return err
 	}
 
-	return WriteProto(c.w, pb)
+	if hasMessage {
```

Comment:
same deal here; I think it's far more likely that it has a message, so I'd let that be the outer case.